src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,957 @@
+        public const int Width = 24;
+
+        /// <summary>
+        /// Number of sensor pixel array rows
+        /// </summary>
+        public const int Height = 32;
+
+        /// <summary>
+        /// Total number of pixels
+        /// </summary>
+        public const int PixelCount = Width * Height;
+
+        private const int OPENAIR_TA_SHIFT = 8;
+        private const double SCALEALPHA = 0.000001;
+
+        private I2cDevice _i2cDevice;
+        private ushort[] _eeData = new ushort[832];
+
+        private int[] _alpha = new int[PixelCount];
+        private double _alphaPTAT = 0;
+        private byte _alphaScale = 0;
+        private ushort[] _brokenPixels = new ushort[5];
+        private int _calibrationModeEE = 0;
+        private float[] _cpAlpha = new float[2];
+        private float _cpKta = 0;
+        private float _cpKv = 0;
+        private int[] _cpOffset = new int[2];
+        private int[] _ct = new int[5];
+        private short _gainEE = 0;
+        private float[] _ilChessC = new float[3];
+        private float _ksTa = 0;
+        private double[] _ksTo = new double[5];
+        private byte[] _kta = new byte[PixelCount];
+        private byte _ktaScale = 0;
+        private float _ktPTAT = 0;
+        private short _kVdd = 0;
+        private sbyte[] _kv = new sbyte[PixelCount];
+        private float _kvPTAT = 0;
+        private byte _kvScale = 0;
+        private int[] _offset = new int[PixelCount];
+        private ushort[] _outlierPixels = new ushort[5];
+        private byte _resolutionEE = 0;
+        private float _tgc = 0;
+        private short _vdd25 = 0;
+        private short _vPTAT25 = 0;
+
+        /// <summary>
+        /// MLX90640 Default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x33;
+
+        /// <summary>
+        /// Creates a new instance of the MLX90640
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Mlx90640(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            DumpEE();
+            ExtractVDDParameters();
+            ExtractPTATParameters();
+            ExtractVDDParameters();
+            ExtractPTATParameters();
+            ExtractGainParameters();
+            ExtractTgcParameters();
+            ExtractResolutionParameters();
+            ExtractKsTaParameters();
+            ExtractKsToParameters();
+            ExtractCPParameters();
+            ExtractAlphaParameters();
+            ExtractOffsetParameters();
+            ExtractKtaPixelParameters();
+            ExtractKvPixelParameters();
+            ExtractCILCParameters();
+            ExtractDeviatingPixels();
+        }
+
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sampling Mode</param>
+        public void SetSampling(Sampling mode)
```

Comment:
Why not create a property with it?
Also, why not directly having the correct values to Sampling? 
like `Sampling_64_Hz = 0x07 << 7`
So you won't have to adjust in the code. You'll need to change the type to ushort (see the comment as well on Write16) for the enum but it doesn't really matter.